package/network/utils/iwinfo/patches/0018-nl80211-add-support-for-HE-htmodes.patch

```diff
@@ -0,0 +1,84 @@
+From ea1dc7d6cca921d78d2e7bcb467d059cd60567ef Mon Sep 17 00:00:00 2001
+From: Andre Heider <a.heider@gmail.com>
+Date: Fri, 18 Nov 2022 12:14:47 +0100
+Subject: [PATCH 18/39] nl80211: add support for HE htmodes
+
+Query hostapd or wpa_supplicant to determine if 802.11ax is enabled as
+nl80211 doesn't offer that info.
+
+Signed-off-by: Andre Heider <a.heider@gmail.com>
+---
+ iwinfo_nl80211.c | 33 ++++++++++++++++++++++++++-------
+ 1 file changed, 26 insertions(+), 7 deletions(-)
+
+diff --git a/iwinfo_nl80211.c b/iwinfo_nl80211.c
+index df244f5..b86e624 100644
+--- a/iwinfo_nl80211.c
++++ b/iwinfo_nl80211.c
+@@ -3272,9 +3272,10 @@ static int nl80211_get_htmode_cb(struct nl_msg *msg, void *arg)
+ 
+ static int nl80211_get_htmode(const char *ifname, int *buf)
+ {
+-	struct chan_info chn = { .width = 0, .mode = 0 };
+-	char *res;
++	struct chan_info chn = { 0 };
++	char *res, b[2] = { 0 };
+ 	int err;
++	bool he = false;
+ 
+ 	res = nl80211_phy2ifname(ifname);
+ 	*buf = 0;
+@@ -3285,27 +3286,45 @@ static int nl80211_get_htmode(const char *ifname, int *buf)
+ 	if (err)
+ 		return -1;
+ 
++	if (nl80211_hostapd_query(res ? res : ifname, "ieee80211ax", b, sizeof(b)))
++		he = b[0] == '1';
++	else if (nl80211_wpactl_query(res ? res : ifname, "wifi_generation", b, sizeof(b)))
++		he = b[0] == '6';
```

Comment:
Where is `b` used?

Reply:
Just for that `bool he`, `b` is just an intermediate since we're getting `char*` results form the socket/config.

Reply:
ok, now I get the code.